File: core-functionality/src/main/java/core/config/CassandraConfig.java
package core.config;

import org.springframework.context.annotation.Configuration;
import org.springframework.data.cassandra.config.AbstractCassandraConfiguration;
import org.springframework.data.cassandra.repository.config.EnableCassandraRepositories;

@Configuration
@EnableCassandraRepositories(basePackages = "core.process.api.repository")
public class CassandraConfig extends AbstractCassandraConfiguration {

    @Override
    protected String getKeyspaceName() {
        return "my_keyspace";
    }

    @Override
    protected String getContactPoints() {
        return "localhost";
    }

    @Override
    protected int getPort() {
        return 9042;
    }
}

File: core-functionality/src/main/kotlin/core/config/CassandraConfig.kt
package core.config

import org.springframework.context.annotation.Configuration
import org.springframework.data.cassandra.config.AbstractCassandraConfiguration
import org.springframework.data.cassandra.repository.config.EnableCassandraRepositories

@Configuration
@EnableCassandraRepositories
class CassandraConfig : AbstractCassandraConfiguration() {
    override fun getKeyspaceName(): String = "my_keyspace"
    override fun getContactPoints(): String = "localhost"
    override fun getPort(): Int = 9042
}